controllers/cluster-registration/cluster_registration_controller.go

```diff
@@ -182,6 +196,71 @@ func (r *RegisteredClusterReconciler) Reconcile(computeContextOri context.Contex
 	return ctrl.Result{}, nil
 }
 
+func (r *RegisteredClusterReconciler) checkSynctargetExists(locationContext context.Context, regCluster *singaporev1alpha1.RegisteredCluster) (int, error) {
+
+	locationClusterName, _ := logicalcluster.ClusterFromContext(locationContext)
+	labels := RegisteredClusterNamelabel + "=" + regCluster.Name + "," + RegisteredClusterNamespacelabel + "=" + regCluster.Namespace + "," + RegisteredClusterWorkspace + "=" + strings.ReplaceAll(locationClusterName.String(), ":", "-")
+
+	syncTargetList, err := r.ComputeDynamicClient.Resource(clusterGVR).List(locationContext, metav1.ListOptions{
+		LabelSelector: labels,
+	})
+	if err != nil {
+		return 0, giterrors.WithStack(err)
+	}
+
+	r.Log.V(4).Info("Number of synctarget found with lables",
+		"number", len(syncTargetList.Items),
+		RegisteredClusterNamelabel, regCluster.Name,
+		RegisteredClusterNamespacelabel, regCluster.Namespace)
+
+	if len(syncTargetList.Items) == 1 {
+		return len(syncTargetList.Items), nil
+	}
+
+	return len(syncTargetList.Items), nil
+
+}
+
+func (r *RegisteredClusterReconciler) syncSyncTarget(computeContext context.Context, regCluster *singaporev1alpha1.RegisteredCluster, managedCluster *clusterapiv1.ManagedCluster) error {
+
+	logger := r.Log.WithName("syncSyncTarget").WithValues("namespace", regCluster.Namespace, "name", regCluster.Name, "managed cluster name", managedCluster.Name)
+
+	if status, ok := helpers.GetConditionStatus(regCluster.Status.Conditions, clusterapiv1.ManagedClusterConditionJoined); ok && status == metav1.ConditionTrue {
+
+		locationContext := logicalcluster.WithCluster(computeContext, logicalcluster.New(regCluster.Spec.Location))
+		locationClusterName, _ := logicalcluster.ClusterFromContext(locationContext)
+
+		syncTargetLength, err := r.checkSynctargetExists(locationContext, regCluster)
+		if err != nil {
+			return giterrors.WithStack(err)
+		}
+		if syncTargetLength == 0 {
+			syncTarget := &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": workloadv1alpha1.SchemeGroupVersion.String(),
+					"kind":       "SyncTarget",
+					"metadata": map[string]interface{}{
+						"generateName": "registered-cluster-",
```

Comment:
Shall we use the registered cluster's name (or some initial portion of it if necessary keeping max length in mind) as the prefix rather than `registered-cluster-`?  I think that will be preferred for usability.